pkg/cluster/tinkerbell.go

```diff
@@ -26,6 +26,24 @@ func tinkerbellEntry() *ConfigManagerEntry {
 			machineConfigsProcessor(processTinkerbellMachineConfig),
 			processTinkerbellTemplateConfigs,
 		},
+		Validations: []Validation{
+			func(c *Config) error {
+				if c.TinkerbellDatacenter != nil {
+					if err := validateSameNamespace(c, c.TinkerbellDatacenter); err != nil {
+						return err
```

Comment:
This is so weird that it is not detecting this, do you happen to know why that might be the case? Like if we add a print statement here, do you see the unit test go through this code path?

Reply:
cuz the test uses cloudstack not tinkerell 😂 fixed now